server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -71,10 +73,14 @@ private void handleUpdate(String mailboxId, MailboxUpdateRequest updateRequest,
         try {
             validateMailboxName(updateRequest, mailboxSession);
             Mailbox mailbox = getMailbox(mailboxId, mailboxSession);
+            checkRole(mailbox.getRole());
```

Comment:
I partially agree : a method should be as short as possible. when you have to handle a lot of preconditions, it doesn't make sense to handle them with if statements. On the other side, I agree that we should not abuse exception for local matters.
The right solution is probably to have a precondition method that return a list of errors, then we would do a if on the error list. But the cost of doing this here is high.
Finally, if I quote your quote : 
> try/catch is for things that go wrong that are outside of your control and not 
> in the normal program flow. 
We can argue that the case we are talking about is not normal program flow.